### ISSUE TITLE
APyFixedArray conversion resulting in wrong number while APyFixed remains correct

### ISSUE THREAD

**Issue by rrbtmp** [apytypes/apytypes#487]
I have an example where creating an APyFixedArray from numpy array seems to produce the wrong result, while APyFixed remains correct. Could you take a look?

```python
from apytypes import APyFixed, APyFixedArray
import numpy as np

fl = 1.2124003311558797e-16
npa = np.array([fl])

fx = APyFixed.from_float(fl, 1, 11)
fl2fxa = APyFixedArray.from_float([fl], 1, 11)
npa2fxa = APyFixedArray.from_array(npa, 1, 11)
npa2fx = APyFixed.from_float(npa[0], 1, 11)

print(
    f"{fl=}\n",
    f"{npa=}\n",
    f"{fx=}\n",
    f"{fl2fxa=}\n",
    f"{npa2fxa=} ({npa2fxa[0].to_bits()})\n",
    f"{npa2fx=} ({npa2fx.to_bits()})\n",
)
```

Notice below how the numpy array to APyFixedArray (**npa2fxa**) diverges from numpy array to APyFixed (**npa2fx**):
```
fl=1.2124003311558797e-16
npa=array([1.21240033e-16])
fx=APyFixed(0, bits=12, int_bits=1)
fl2fxa=APyFixedArray([0], shape=(1,), bits=12, int_bits=1)
npa2fxa=APyFixedArray([990], shape=(1,), bits=12, int_bits=1) (990)
npa2fx=APyFixed(0, bits=12, int_bits=1) (0)
```

Test on Python 3.9.17, both on both APyTypes v0.1.0 and the latest commit (3f6f90bb6e81e08166b8d262baae17f911dd044b).

**Comment by miklhh** [comment]
Hi @rrbtmp,

Thanks for submitting this issue. I will look into it.

**Comment by miklhh** [comment]
Thanks for submitting this bug @rrbtmp. It has been fixed in #488.

```python3
>>> from apytypes import APyFixed, APyFixedArray
>>> import numpy as np
>>>
>>> fl = 1.2124003311558797e-16
>>> npa = np.array([fl])
>>> 
>>> fx = APyFixed.from_float(fl, 1, 11)
>>> fl2fxa = APyFixedArray.from_float([fl], 1, 11)
>>> npa2fxa = APyFixedArray.from_array(npa, 1, 11)
>>> npa2fx = APyFixed.from_float(npa[0], 1, 11)
>>>
>>> print(
>>>     f"{fl=}\n",
>>>     f"{npa=}\n",
>>>     f"{fx=}\n",
>>>     f"{fl2fxa=}\n",
>>>     f"{npa2fxa=} ({npa2fxa[0].to_bits()})\n",
>>>     f"{npa2fx=} ({npa2fx.to_bits()})\n",
>>> )

fl=1.2124003311558797e-16
npa=array([1.21240033e-16])
fx=APyFixed(0, bits=12, int_bits=1)
fl2fxa=APyFixedArray([0], shape=(1,), bits=12, int_bits=1)
npa2fxa=APyFixedArray([0], shape=(1,), bits=12, int_bits=1) (0)
npa2fx=APyFixed(0, bits=12, int_bits=1) (0)
```

**Comment by rrbtmp** [comment]
Thank you very much for the fix, cheers.
